numpy/core/tests/test_records.py

```diff
@@ -12,7 +12,7 @@
 import pytest
 
 import numpy as np
-from numpy.compat import Path
+from pathlib import Path
```

Comment:
@eric-wieser  Do you want to keep changes like this? That modernize the test suite so as not to use our own compatibility layer?

I feel like if we remove these, we are no longer testing the compatibility layer, and it is as good as removing it (since bug will creep in that we won't be monitoring).

Reply:
If we want to test the compatibility layer, we should have a standalone test for that. I don't think the lack of that test should motivate us not to clean up our other tests.

Reply:
but as it stands, there are no other tests dedicated for the compatibility layer. 

I would prefer to wait unit you drop 3.5 support. then the only test becomes

```
assert os_PathLike == os.PathLike
[...]
```